src/main/java/org/truffleruby/language/methods/ModuleBodyDefinitionNode.java

```diff
@@ -74,7 +74,13 @@ private LexicalScope prepareLexicalScope(LexicalScope staticLexicalScope, Lexica
         } else {
             // Cache the scope per module in case the module body is run multiple times.
             // This allows dynamic constant lookup to cache better.
-            return lexicalScopes.computeIfAbsent(module, m -> new LexicalScope(parentLexicalScope, module));
+            LexicalScope scope = lexicalScopes.get(module);
```

Comment:
btw: It's possible to configure idea to specially highlight reassigned local variables and arguments. (I have red underlines.)